packages/algob/src/lib/script-checkpoints.ts

```diff
@@ -81,23 +82,43 @@ export class CheckpointRepoImpl implements CheckpointRepo {
   strippedCP: Checkpoints = {};
   precedingCP: Checkpoints = {};
   allCPs: Checkpoints = {};
+  scriptMap: assetScriptMap = {};
 
-  private _mergeTo (target: Checkpoints, cp: Checkpoints): Checkpoints {
+  private _mergeTo (target: Checkpoints, cp: Checkpoints, scriptMap: assetScriptMap): Checkpoints {
     const keys: string[] = Object.keys(cp);
     return keys.reduce((out: Checkpoints, key: string) => {
       return appendToCheckpoint(out, key, cp[key]);
     }, target);
   }
 
-  merge (cp: Checkpoints): CheckpointRepo {
+  merge (cp: Checkpoints, scriptName: string): CheckpointRepo {
     this.strippedCP = cp;
-    this.precedingCP = this._mergeTo(this.precedingCP, cp);
-    this.mergeToGlobal(cp);
+    this.precedingCP = this._mergeTo(this.precedingCP, cp, this.scriptMap);
+    this.mergeToGlobal(cp, scriptName);
     return this;
   }
 
-  mergeToGlobal (cp: Checkpoints): CheckpointRepo {
-    this.allCPs = this._mergeTo(this.allCPs, cp);
+  mergeToGlobal (cp: Checkpoints, scriptName: string): CheckpointRepo {
+    this.allCPs = this._mergeTo(this.allCPs, cp, this.scriptMap);
+
+    if (scriptName.charAt(0) === 'a') {
+      scriptName = scriptName.slice(10, scriptName.length - 8);
+    }
+
+    const keys: string[] = Object.keys(cp);
+    for (const k of keys) {
+      const orig = cp[k];
+      const allAssetNames = Object.keys(orig.asa).concat(Object.keys(orig.asc));
+      for (const assetName of allAssetNames) {
+        if (!(this.scriptMap[assetName])) { this.scriptMap[assetName] = scriptName; } else {
```

Comment:
Can you expand this statement?
This already has three levels of nesting so custom formatting is even worse.
Also it creates a long line.

Reply:
yes